### PR TITLE
[COTCO-639][LB] added code style checker to BE

### PR DESCRIPTION
**Pull request by udinobi**

### CODE REVIEW

File: scalastyle-config.xml [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+      Are you sure that you want to throw Error? In most cases, you should use appropriate Exception instead.
+      If you must throw Error, wrap the code block with
+      // scalastyle:off throwerror
+      throw new XXXError(...)
+      // scalastyle:on throwerror
+    ]]></customMessage>
+  </check>
+
+  <check level="warning" class="org.scalastyle.scalariform.ClassNamesChecker" enabled="true">
+    <parameters><parameter name="regex"><![CDATA[[A-Z][A-Za-z]*]]></parameter></parameters>
```
Comment (greghmrc) [inline scalastyle-config.xml]: That rules out classes like Tuple22 or test classes with numbers like ShouldGive123When567Happens (just an example, but you got the idea).

File: scalastyle-config.xml [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+    </parameters>
+  </check>
+
+  <check level="warning" class="org.scalastyle.scalariform.MagicNumberChecker" enabled="true">
+    <parameters><parameter name="ignore">-1,0,1,2,3</parameter></parameters>
+  </check>
+
+  <check level="warning" class="org.scalastyle.scalariform.MethodNamesChecker" enabled="true">
+    <parameters>
+    <parameter name="regex"><![CDATA[^[a-z][A-Za-z0-9]*$]]></parameter>
```
Comment (greghmrc) [inline scalastyle-config.xml]: What do you think about including underscore?
https://docs.scala-lang.org/style/naming-conventions.html#methods

File: scalastyle-config.xml [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    </parameters>
+  </check>
+
+  <check level="warning" class="org.scalastyle.scalariform.MultipleStringLiteralsChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.NoFinalizeChecker" enabled="true"/>
+
+  <check level="error" class="org.scalastyle.scalariform.NonASCIICharacterChecker" enabled="true"/>
```
Comment (greghmrc) [inline scalastyle-config.xml]: That rules out Intellij config that renders nice right arrow. Come on even vim supports UTF. Scala compiler definitely does not complain.
And we do not have anyone who uses some creepy text editor from 60's.

File: scalastyle-config.xml [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+  <check level="warning" class="org.scalastyle.scalariform.NoWhitespaceAfterLeftBracketChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.NumberOfMethodsInTypeChecker" enabled="true">
+    <parameters><parameter name="maxMethods"><![CDATA[30]]></parameter></parameters>
+  </check>
+
+  <check level="error" class="org.scalastyle.scalariform.NullChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.ObjectNamesChecker" enabled="true">
+    <parameters><parameter name="regex"><![CDATA[(config|[A-Z][A-Za-z]*)]]></parameter></parameters>
```
Comment (greghmrc) [inline scalastyle-config.xml]: Same as for classes.

File: scalastyle-config.xml [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+  </check>
+
+  <check level="warning" class="org.scalastyle.scalariform.OverrideJavaChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.PackageObjectNamesChecker" enabled="true">
+    <parameters><parameter name="regex"><![CDATA[^[a-z][A-Za-z]*$]]></parameter></parameters>
```
Comment (greghmrc) [inline scalastyle-config.xml]: Underscore is a proper character for a package name. Do we really don't like it? I think it is used in tests at least.

File: scalastyle-config.xml [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+
+  <check level="warning" class="org.scalastyle.scalariform.ProcedureDeclarationChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.PublicMethodsHaveTypeChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.RedundantIfChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.SpaceAfterCommentStartChecker" enabled="true"/>
+
+  <check level="warning" class="org.scalastyle.scalariform.SpacesAfterPlusChecker" enabled="true"/>
```
Comment (greghmrc) [inline scalastyle-config.xml]: How does `Plus Checkers` work with `+:` or `+=` and others like that?

Comment (udinobi) [reply scalastyle-config.xml]: It's ok.

**Pull request merged**